Classes/Security/Authentication/Provider/NeosBackendLdapProvider.php

```diff
@@ -24,54 +25,53 @@
  */
 class NeosBackendLdapProvider extends LdapProvider
 {
-
     /**
-     * @var CompilingEvaluator
      * @Flow\Inject
+     * @var CompilingEvaluator
      */
     protected $eelEvaluator;
 
     /**
-     * Create a new account for the given credentials. Return null if you
-     * do not want to create a new account, that is, only authenticate
-     * existing accounts from the database and fail on new logins.
-     *
-     * @param array $credentials array containing username and password
-     * @param array $ldapSearchResult
-     * @return Account
+     * @Flow\Inject
+     * @var UserService
+     */
+    protected $userService;
+
+    /**
+     * @inheritdoc
      */
-    protected function createAccount(array $credentials, array $ldapSearchResult)
+    protected function createAccount(array $credentials, array $ldapUserData)
     {
         $mapping = Arrays::arrayMergeRecursiveOverrule(
             [
                 'firstName' => 'user.givenName[0]',
-                'lastName' => 'user.sn[0]'
+                'lastName' => 'user.sn[0]',
             ],
-            isset($this->options['mapping']) ? $this->options['mapping'] : []
+            $this->options['mapping'] ?? []
         );
-
-        $eelContext = new Context(['user' => $ldapSearchResult]);
+        $eelContext = new Context(['user' => $ldapUserData]);
 
         try {
             $firstName = $this->eelEvaluator->evaluate($mapping['firstName'], $eelContext);
-            $lastName = $this->eelEvaluator->evaluate($mapping['lastName'], $eelContext);
         } catch (\Exception $exception) {
-            // todo : add logging
+            // todo: logging
             $firstName = 'none';
+        }
+        try {
+            $lastName = $this->eelEvaluator->evaluate($mapping['lastName'], $eelContext);
+        } catch (\Exception $exception) {
+            // todo: logging
             $lastName = 'none';
         }
 
-        $userService = new \Neos\Neos\Domain\Service\UserService();
-        $user = $userService->createUser(
+        $user = $this->userService->createUser(
```

Comment:
I think we should get rid of the usage of the `UserService`.

That's seems to be the only reason for the Neos dependency which makes the package unusable for Flow apps.

Also the `createUser` method will even create a personal workspace or might even assume the role `Neos.Neos:Editor` which is also not intended for frontend users.

I think we can replace that with a custom method in this class that could offer a way to still do those steps if needed in an app.